apps/wolfsshd/configuration.c

```diff
@@ -0,0 +1,908 @@
+            conf->permitRootLogin = 0;
+        }
+        else if (WSTRCMP(value, "yes") == 0) {
+            conf->permitRootLogin = 1;
+        }
+        else {
+            ret = WS_BAD_ARGUMENT;
+        }
+    }
+
+    return ret;
+}
+
+/* returns WS_SUCCESS on success */
+static int HandlePwAuth(WOLFSSHD_CONFIG* conf, const char* value)
+{
+    int ret = WS_SUCCESS;
+
+    if (conf == NULL || value == NULL) {
+        ret = WS_BAD_ARGUMENT;
+    }
+
+    if (ret == WS_SUCCESS) {
+        if (WSTRCMP(value, "no") == 0) {
+            conf->passwordAuth = 0;
+        }
+        else if (WSTRCMP(value, "yes") == 0) {
+            conf->passwordAuth = 1;
+        }
+        else {
+            ret = WS_BAD_ARGUMENT;
+        }
+    }
+
+    return ret;
+}
+
+#define WOLFSSH_PROTOCOL_VERSION 2
+
+/* returns WS_SUCCESS on success */
+static int HandleProtocol(WOLFSSHD_CONFIG* conf, const char* value)
+{
+    int ret = WS_SUCCESS;
+    long portInt;
+
+    if (conf == NULL || value == NULL) {
+        ret = WS_BAD_ARGUMENT;
+    }
+
+    if (ret == WS_SUCCESS) {
+        portInt = GetConfigInt(value, (int)WSTRLEN(value), 0, conf->heap);
+        if (portInt <= 0) {
+            wolfSSH_Log(WS_LOG_ERROR, "[SSHD] Invalid protocol number: %s.",
+                        value);
+            ret = WS_BAD_ARGUMENT;
+        }
+        else {
+            if (portInt != WOLFSSH_PROTOCOL_VERSION) {
+                wolfSSH_Log(WS_LOG_ERROR, "[SSHD] Protocol number %ld not "
+                    "supported.", portInt);
+                ret = WS_BAD_ARGUMENT;
+            }
+        }
+    }
+
+    return ret;
+}
+
+/* returns WS_SUCCESS on success */
+static int HandlePort(WOLFSSHD_CONFIG* conf, const char* value)
+{
+    int ret = WS_SUCCESS;
+    long portInt;
+
+    if (conf == NULL || value == NULL) {
+        ret = WS_BAD_ARGUMENT;
+    }
+
+    if (ret == WS_SUCCESS) {
+        portInt = GetConfigInt(value, (int)WSTRLEN(value), 0, conf->heap);
+        if (portInt <= 0) {
+            wolfSSH_Log(WS_LOG_ERROR, "[SSHD] Invalid port number: %s.",
+                        value);
+            ret = WS_BAD_ARGUMENT;
+        }
+        else {
+            if (portInt <= (word16)-1) {
+                conf->port = (word16)portInt;
+            }
+            else {
+                wolfSSH_Log(WS_LOG_ERROR, "[SSHD] Port number %ld too big.",
+                            portInt);
+                ret = WS_BAD_ARGUMENT;
+            }
+        }
+    }
+
+    return ret;
+}
+
+static int HandleInclude(WOLFSSHD_CONFIG *conf, const char *value)
+{
+    const char *ptr;
+    const char *ptr2;
+    const char *postfix = NULL;
+    const char *prefix = NULL;
+    int prefixLen = 0;
+    int found = 0;
+
+    /* No value, nothing to do */
+    if (!value || value[0] == '\0') {
+        return WS_BAD_ARGUMENT;
+    }
+
+    /* Ignore trailing whitespace */
+    ptr = value + WSTRLEN(value) - 1;
+    while (ptr != value) {
+        if (!isspace(*ptr)) {
```

Comment:
This is more a question. We use wrappers for malloc and free, and bunch of the string functions. Do we want to be mixing the macro versions and bare calls? Is this expected to be built for Linux only, or for other OSes?

Reply:
if using the isspace we should abstract it to a macro, will fix it